crashes in CI when running `scala-cli fmt --check`

**Version(s)**
1.12.5, via https://github.com/VirtusLab/scala-cli-setup/releases/tag/v1.12.5
previously 1.9.1, via https://github.com/VirtusLab/scala-cli-setup/releases/tag/v1.9.1

**Describe the bug**
I have a simple GitHub Actions job that runs scalafmt (with an older version of scala-cli):
```
  scalafmt:
    runs-on: ubuntu-latest
    steps:
      - name: Checkout
        uses: actions/checkout@v5
      - uses: coursier/cache-action@bebeeb0e6f48ebad66d3783946588ecf43114433 # v6.4.8
      - name: scala-cli-setup-action
        uses: VirtusLab/scala-cli-setup@77834b5926f3eb70869d8009530c65585f7a039b # v1.9.1
      - name: scalafmtCheckAll
        run: scala-cli fmt --check
```

This would occasionally segfault:
```
Run scala-cli fmt --check
  scala-cli fmt --check
  shell: /usr/bin/bash -e {0}
  env:
    JAVA_HOME: /home/runner/.cache/coursier/arc/https/github.com/adoptium/temurin17-binaries/releases/download/jdk-17.0.16%252B8/OpenJDK17U-jdk_x64_linux_hotspot_17.0.16_8.tar.gz/jdk-17.0.16+8
    COURSIER_BIN_DIR: /home/runner/cs/bin
Looking for unformatted files...
Looking for unformatted files... [          ]
  out 0.0% 0, fmt 0.0% 0, in 0.0% 0 todo=283
ScalaNative: Unrecoverable NullPointerException in user thread
/home/runner/work/_temp/c1fb3d54-ead7-4b0d-8640-a5d2dc86d101.sh: line 1:  2194 Segmentation fault      (core dumped) scala-cli fmt --check
Error: Process completed with exit code 139.
```

I updated to the latest scala-cli:
```
        uses: VirtusLab/scala-cli-setup@14e5155ee896fcc200f66f92c9eb7a3dd9b58e0d # v1.12.5
```
and now seeing a new occasional failure (only once so far):
```
Run scala-cli fmt --check
  scala-cli fmt --check
  shell: /usr/bin/bash -e {0}
  env:
    JAVA_HOME: /home/runner/.cache/coursier/arc/https/github.com/adoptium/temurin17-binaries/releases/download/jdk-17.0.16%252B8/OpenJDK17U-jdk_x64_linux_hotspot_17.0.16_8.tar.gz/jdk-17.0.16+8
    COURSIER_BIN_DIR: /home/runner/cs/bin
Looking for unformatted files...
Looking for unformatted files... [          ]
  out 0.0% 0, fmt 0.0% 0, in 0.0% 0 todo=284
java.lang.ClassCastException: class scala.scalanative.unsafe.CFuncPtr1 cannot be cast to scala.scalanative.runtime.ByteArray
	at scala.scalanative.runtime.package$.throwClassCast(Unknown Source)
	at java.nio.file.Files$.$anonfun$readAllBytes$1(Unknown Source)
	at java.nio.file.Files$$$Lambda$10.apply(Unknown Source)
	at scala.scalanative.unsafe.Zone$.acquire(Unknown Source)
	at org.scalafmt.sysops.GranularPlatformAsyncOps$.$anonfun$readFileAsync$1(Unknown Source)
	at org.scalafmt.sysops.GranularPlatformAsyncOps$$$Lambda$1.apply(Unknown Source)
	at scala.concurrent.Future$$$Lambda$1.apply(Unknown Source)
	at scala.concurrent.impl.Promise$Transformation.run(Unknown Source)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
	at java.lang.Thread.run(Unknown Source)
	at scala.scalanative.runtime.NativeThread$.threadEntryPoint(Unknown Source)
	at <none>._SM49scala.scalanative.runtime.NativeThread$$$Lambda$1G17$extern$forwarder(Unknown Source)
	at <none>.(Unknown Source)
	at <none>.(Unknown Source)
	at <none>.(Unknown Source)
Looking for unformatted files... [          ]
  out 0.7% 2, fmt 0.7% 2, in 1.1% 3/1 todo=280
error: UnexpectedError=8
Error: Process completed with exit code 8.
```

**To Reproduce**
Unfortunately the code being formatted is a private repository (and the issue is infrequent). The segfault occurred roughly once in every 20-100 runs. Since updating to the latest scala-cli, I witnessed the `ClassCastException` just a single time (in a total of maybe 10 runs).

**Expected behaviour**
I expect `scala-cli fmt --check` to not crash

just had this happen again:
```
Run scala-cli fmt --check
Looking for unformatted files...
Looking for unformatted files... [          ]
  out 0.0% 0, fmt 0.0% 0, in 0.0% 0 todo=299
Looking for unformatted files... [#####     ]
  out 58.2% 174, fmt 58.2% 174, in 58.9% 176 todo=123
java.lang.ClassCastException: class java.lang.Integer cannot be cast to scala.scalanative.runtime.ByteArray
Looking for unformatted files... [#######   ]
  out 71.6% 214, fmt 71.6% 214, in 71.9% 215/1 todo=83
error: UnexpectedError=8
	at scala.scalanative.runtime.package$.throwClassCast(Unknown Source)
	at java.nio.file.Files$.$anonfun$readAllBytes$1(Unknown Source)
	at java.nio.file.Files$$$Lambda$10.apply(Unknown Source)
	at scala.scalanative.unsafe.Zone$.acquire(Unknown Source)
	at org.scalafmt.sysops.GranularPlatformAsyncOps$.$anonfun$readFileAsync$1(Unknown Source)
	at org.scalafmt.sysops.GranularPlatformAsyncOps$$$Lambda$1.apply(Unknown Source)
	at scala.concurrent.Future$$$Lambda$1.apply(Unknown Source)
	at scala.concurrent.impl.Promise$Transformation.run(Unknown Source)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
	at java.lang.Thread.run(Unknown Source)
	at scala.scalanative.runtime.NativeThread$.threadEntryPoint(Unknown Source)
	at <none>._SM49scala.scalanative.runtime.NativeThread$$$Lambda$1G17$extern$forwarder(Unknown Source)
	at <none>.(Unknown Source)
	at <none>.(Unknown Source)
	at <none>.(Unknown Source)
Error: Process completed with exit code 8.
```

using `VirtusLab/scala-cli-setup@14e5155ee896fcc200f66f92c9eb7a3dd9b58e0d`, it's showing `scala-cli` version `1.12.5`:
```
Install Apps
  Setting COURSIER_BIN_DIR to: /home/runner/cs/bin
  /opt/hostedtoolcache/cs/2.1.25-M23/x64/cs install --contrib sbtn scala-cli:1.12.5
  Wrote sbtn
  Wrote scala-cli
  /home/runner/cs/bin/scala-cli version --cli-version
  1.12.5
```

in my repo, my `.scalafmt.conf` sets `version = 3.11.0`

--- a/project/deps/package.mill
+++ b/project/deps/package.mill
@@ -134,7 +134,7 @@ object Deps {
     def scalaMeta                         = "4.15.2"
     def scalafmt                          = "3.11.0"
     def scalaNative04                     = "0.4.17"
-    def scalaNative05                     = "0.5.10"
+    def scalaNative05                     = "0.5.11"
     def scalaNative                       = scalaNative05
     def maxScalaNativeForToolkit          = scalaNative05
     def maxScalaNativeForTypelevelToolkit = scalaNative04

--- a/website/docs/reference/cli-options.md
+++ b/website/docs/reference/cli-options.md
@@ -1452,7 +1452,7 @@ Enable Scala Native. To show more options for Scala Native pass `--help-native`
 
 ### `--native-version`
 
-Set the Scala Native version (0.5.10 by default).
+Set the Scala Native version (0.5.11 by default).
 
 ### `--native-mode`
 

--- a/website/docs/reference/directives.md
+++ b/website/docs/reference/directives.md
@@ -553,7 +553,7 @@ Add Scala Native options
 
 `//> using nativeLto full`
 
-`//> using nativeVersion 0.5.10`
+`//> using nativeVersion 0.5.11`
 
 `//> using nativeCompile -flto=thin`
 

--- a/website/docs/reference/scala-command/cli-options.md
+++ b/website/docs/reference/scala-command/cli-options.md
@@ -904,7 +904,7 @@ Enable Scala Native. To show more options for Scala Native pass `--help-native`
 
 `SHOULD have` per Scala Runner specification
 
-Set the Scala Native version (0.5.10 by default).
+Set the Scala Native version (0.5.11 by default).
 
 ### `--native-mode`
 

--- a/website/docs/reference/scala-command/directives.md
+++ b/website/docs/reference/scala-command/directives.md
@@ -350,7 +350,7 @@ Add Scala Native options
 
 `//> using nativeLto full`
 
-`//> using nativeVersion 0.5.10`
+`//> using nativeVersion 0.5.11`
 
 `//> using nativeCompile -flto=thin`
 

--- a/website/docs/reference/scala-command/runner-specification.md
+++ b/website/docs/reference/scala-command/runner-specification.md
@@ -178,7 +178,7 @@ Enable Scala Native. To show more options for Scala Native pass `--help-native`
 
 **--native-version**
 
-Set the Scala Native version (0.5.10 by default).
+Set the Scala Native version (0.5.11 by default).
 
 **--native-mode**
 
@@ -992,7 +992,7 @@ Enable Scala Native. To show more options for Scala Native pass `--help-native`
 
 **--native-version**
 
-Set the Scala Native version (0.5.10 by default).
+Set the Scala Native version (0.5.11 by default).
 
 **--native-mode**
 
@@ -1605,7 +1605,7 @@ Enable Scala Native. To show more options for Scala Native pass `--help-native`
 
 **--native-version**
 
-Set the Scala Native version (0.5.10 by default).
+Set the Scala Native version (0.5.11 by default).
 
 **--native-mode**
 
@@ -2244,7 +2244,7 @@ Enable Scala Native. To show more options for Scala Native pass `--help-native`
 
 **--native-version**
 
-Set the Scala Native version (0.5.10 by default).
+Set the Scala Native version (0.5.11 by default).
 
 **--native-mode**
 
@@ -2902,7 +2902,7 @@ Enable Scala Native. To show more options for Scala Native pass `--help-native`
 
 **--native-version**
 
-Set the Scala Native version (0.5.10 by default).
+Set the Scala Native version (0.5.11 by default).
 
 **--native-mode**
 
@@ -3536,7 +3536,7 @@ Enable Scala Native. To show more options for Scala Native pass `--help-native`
 
 **--native-version**
 
-Set the Scala Native version (0.5.10 by default).
+Set the Scala Native version (0.5.11 by default).
 
 **--native-mode**
 
@@ -4207,7 +4207,7 @@ Enable Scala Native. To show more options for Scala Native pass `--help-native`
 
 **--native-version**
 
-Set the Scala Native version (0.5.10 by default).
+Set the Scala Native version (0.5.11 by default).
 
 **--native-mode**
 
@@ -4938,7 +4938,7 @@ Enable Scala Native. To show more options for Scala Native pass `--help-native`
 
 **--native-version**
 
-Set the Scala Native version (0.5.10 by default).
+Set the Scala Native version (0.5.11 by default).
 
 **--native-mode**
 
@@ -5925,7 +5925,7 @@ Enable Scala Native. To show more options for Scala Native pass `--help-native`
 
 **--native-version**
 
-Set the Scala Native version (0.5.10 by default).
+Set the Scala Native version (0.5.11 by default).
 
 **--native-mode**
 

